dynamic-property-zk/src/main/java/ru/fix/dynamic/property/zk/ZkDynamicPropertySource.java

```diff
@@ -0,0 +1,274 @@
+        String path = getAbsolutePath(key);
+        ChildData currentData = treeCache.getCurrentData(path);
+        return currentData == null ? defaulValue : new String(currentData.getData(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public <T> T getProperty(String key, Class<T> type) {
+        return getProperty(key, type, null);
+    }
+
+    @Override
+    public <T> T getProperty(String key, Class<T> type, T defaultValue) {
+        String value = getProperty(key);
+        if (value != null) {
+            return marshaller.unmarshall(value, type);
+        }
+        return defaultValue;
+    }
+
+    /**
+     * Works through curator directly to load latest data
+     */
+    @Override
+    public Properties getAllProperties() throws Exception {
+        Properties allProperties = new Properties();
+        Stat exist = curatorFramework.checkExists().forPath(getAbsolutePath(""));
+        if (exist != null) {
+            List<String> childs = curatorFramework.getChildren().forPath(getAbsolutePath(""));
+            if (!childs.isEmpty()) {
+                CountDownLatch latcher = new CountDownLatch(childs.size());
+                for (String child : childs) {
+                    curatorFramework.getData().watched().inBackground((client, event) -> {
+                        allProperties.put(child, new String(event.getData(), StandardCharsets.UTF_8));
+                        latcher.countDown();
+                    }).forPath(getAbsolutePath(child));
+                }
+                if (!latcher.await(120, TimeUnit.SECONDS)) {
+                    throw new TimeoutException("Failed to extract zk properties data");
+                }
+            }
+        }
+        return allProperties;
+    }
+
+    /**
+     * Works through curator directly to load latest data
+     */
+    @Override
+    @SuppressWarnings("squid:S1166") // Either log or rethrow this exception.
+    public Map<String, String> getAllSubtreeProperties(String root) throws Exception {
+        Map<String, String> allProperties = new ConcurrentHashMap<>();
+        Stat exist = curatorFramework.checkExists().forPath(getAbsolutePath(root));
+        if (exist != null) {
+            getProperty(allProperties, null == root ? "" : root);
+        }
+        return allProperties;
+    }
+
+    private void getProperty(Map<String, String> allProperties, String path) {
+        Map<String, ChildData> currentChildren = treeCache.getCurrentChildren(getAbsolutePath(path));
+        if (currentChildren != null) {
+            currentChildren.forEach((s, childData) -> {
+                String currNodePath = path.isEmpty() ? s : path + '/' + s;
+                allProperties.put(currNodePath, new String(childData.getData(), StandardCharsets.UTF_8));
+                getProperty(allProperties, currNodePath);
+            });
+        }
+    }
+
+    @Override
+    public void updateProperty(String key, String value) throws Exception {
```

Comment:
uploading default property value back to zk with documentation  probably  was bad idea. 
Suggestion: only read from zk, do not upload anything there.

Reply:
remove all unused methods. Now only can put default value or from zk, update and upsert are prohibited